tests/integration/pagure/test_pull_requests.py

```diff
@@ -124,7 +126,11 @@ def test_pr_diff(self):
         assert isinstance(diff, dict)
         assert "README.md" in diff
 
-    def test_pr_diff_empty_diff(self):
-        diff = self.ogr_project.get_pr_files_diff(6)
+    def test_pr_diff_failing(self):
+        with pytest.raises(PagureAPIException):
+            self.ogr_project.get_pr_files_diff(6)
+
+    def test_pr_diff_failing_and_succeding(self):
+        diff = self.ogr_project.get_pr_files_diff(7, retries=1, wait_seconds=0.1)
```

Comment:
Perhaps we could also use flexmock to make sure that the API call is really done twice:

```suggestion
        flexmock(self.ogr_project).should_receive("_call_project_api").twice()
        diff = self.ogr_project.get_pr_files_diff(7, retries=1, wait_seconds=0.1)
```

Reply:
I tried this but if I mock the project I don't receive anymore the dictionary and the exception... probably this should be checked by requere. If we have two requests, requere should check that they both are used. However I debugged the method and I have seen that both requests are used.

Reply:
Ok, let's not comoplicate this.